core/client/meetinthemiddle.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2020 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+// Meet in the middle algorithm:
+//
+
+// NewerRevisionsToVerify returns a list of revisions to perform append-only verifications against.
+// The relying party / sender is assured that if there is an entity running the
+// OlderRevisionsToVerify algorithm, they will have at least on revision in
+// common.
+//
+// `created` is the (earliest) revision a data item was found in the map.
+//    It is OK to use the revision at which an item was fetched, but the algorithm will be more efficent
+//    if the algorithm starts with the revision at which an item was submitted to the map.
+// `verified` is the latest revision of a successfully verified consistency proof by this client.
+//    When verified > created, previously verified revisions are omitted from the results.
+// `current` is the current revision of the map. Selected revisions will be <= current.
+func NewerRevisionsToVerify(created, current, verified int64) []int64 {
+	ret := []int64{} // bⁱ = b-b mod 2ⁱ+ 2ⁱ
+	if created < 1 {
+		created = 1 // Revision 0 is empty.
+	}
+	if current < 1 {
+		return []int64{} // Nothing to verify if the map is empty.
+	}
+	if created > current {
+		created = current
+	}
```

Comment:
This case seems weird. It almost feels like we should panic or return error here.

According to function comment, `created` is the earliest version where an item was found. If `current < created`, we don't know if the value is present in `current` (otherwise `created` would be equal to `current` by definition). Moving it backwards in time may lead to "losing" this data item, and verifying that some irrelevant revisions are consistent instead of the one we are interested in.

Reply:
This is a bit of a special case to help make randomized testing easier. 
My thought was that this would "check" the latest version if `created > current`

Reply:
added panic